mirgecom/eos.py

```diff
@@ -233,3 +232,236 @@ def total_energy(self, cv, pressure):
         """
         return (pressure / (self._gamma - 1.0)
                 + self.kinetic_energy(cv))
+
+
+class PyrometheusMixture(GasEOS):
+    r"""Ideal gas mixture ($p = \rho{R}_\mathtt{mix}{T}$).
+
+    This is the :mod:`pyrometheus`-based EOS. Please refer to the :any:`documentation
+    of Pyrometheus <pyrometheus>` for underlying implementation details.
+
+    Each interface call expects that the :class:`mirgecom.fluid.ConservedVars` object
+    representing the simulation conserved quantities contains at least the
+    canonical conserved quantities mass ($\rho$), energy ($\rho{E}$), and
+    momentum ($\rho\vec{V}$), and the vector of species masses, ($\rho{Y}_\alpha$).
+
+    .. important::
+        When using this EOS, users should take care to match solution initialization
+        with the appropriate units that are used in the user-provided `Cantera`
+        mechanism input files.
+
+    .. automethod:: __init__
+    .. automethod:: get_density
+    .. automethod:: get_species_molecular_weights
+    .. automethod:: get_production_rates
+    .. automethod:: get_species_source_terms
+    .. automethod:: get_internal_energy
+    .. automethod:: species_fractions
+    .. automethod:: total_energy
+    .. automethod:: gamma
+    .. automethod:: gas_const
+
+    Inherits from (and implements) :class:`GasEOS`.
+    """
+
+    def __init__(self, pyrometheus_mech, tguess=300.0):
```

Comment:
```suggestion
    def __init__(self, pyrometheus_mech, temperature_guess=300.0):
```

Reply:
I made this change, and the required example change in (3bc8710).